autosubmitconfigparser/config/configcommon.py

```diff
@@ -1007,7 +1015,7 @@ def _substitute_keys(
         :return: A tuple containing the updated processed dynamic variables and parameters.
         :rtype: Tuple[Dict[str, Any], Dict[str, Any]]
         """
-        for i, key in enumerate(keys):
+        for i, key in enumerate(filter(None, keys)):
```

Comment:
This was added in a separate commit, trying to fix an error that happened in the test experiment with this fix. I will have to write a test for this too. @LuiggiTenorioK I am now trying to replicate the case from your review and then will update the previous commit + test, and then write the test for this, and then test a small experiment. If everything goes well will ask for your review again :+1: 

@mcastril it might take a bit longer to fix this…